src/cryptsetup/cryptsetup-generator.c

```diff
@@ -832,42 +875,7 @@ static int add_crypttab_devices(void) {
                         continue;
                 }
 
-                uuid = startswith(device, "UUID=");
-                if (!uuid)
-                        uuid = path_startswith(device, "/dev/disk/by-uuid/");
-                if (!uuid)
-                        uuid = startswith(name, "luks-");
-                if (uuid)
-                        d = hashmap_get(arg_disks, uuid);
-
-                if (arg_allow_list && !d) {
-                        log_info("Not creating device '%s' because it was not specified on the kernel command line.", name);
-                        continue;
-                }
-
-                r = split_locationspec(keyspec, &keyfile, &keydev);
-                if (r < 0)
-                        return r;
-
-                if (options && (!d || !d->options)) {
-                        r = filter_header_device(options, &headerdev, &filtered_header);
-                        if (r < 0)
-                                return r;
-                        free_and_replace(options, filtered_header);
-                }
-
-                r = create_disk(name,
-                                device,
-                                keyfile,
-                                keydev,
-                                (d && d->options) ? d->headerdev : headerdev,
-                                (d && d->options) ? d->options : options,
-                                arg_crypttab);
-                if (r < 0)
-                        return r;
-
-                if (d)
-                        d->create = false;
+                r = add_crypttab_device(name, device, keyspec, options);
```

Comment:
Missing error handling in the first commit.

Reply:
Fixed.